x-pack/plugins/ml/public/application/explorer/actions/load_explorer_data.ts

```diff
@@ -144,7 +129,7 @@ const loadExplorerDataProvider = (
     const selectionInfluencers = getSelectionInfluencers(selectedCells, viewBySwimlaneFieldName);
     const jobIds = getSelectionJobIds(selectedCells, selectedJobs);
 
-    const bounds = timefilter.getBounds();
+    const bounds = timefilter.getBounds() as Required<TimeRangeBounds>;
```

Comment:
If I just change the time range using the global time picker, the data in the view isn't updating. It only uses the new time range I change something like the swim lane slider.

Reply:
Also doesn't update if using the job selector with 'Apply time range'.

Reply:
Fixed in [add time bounds subject](https://github.com/elastic/kibana/pull/126274/commits/ee8d28e11f7e0cf2c8c4bd0743fcc124cc06bbf2)